include/instances.h

```diff
@@ -0,0 +1,57 @@
+#ifndef INSTANCES_H
+#define INSTANCES_H
+
+#include "ray.h"
+#include "primitive.h"
+#include "quad_primitive.h"
+
+class Instance {
+  public:
+  float* transform; // assumes float[12]
```

Comment:
could probably make a custom coord struct with an x, y, z field, and then create an array if coords instead for greater clarity